fvm/programs.go

```diff
@@ -0,0 +1,92 @@
+package fvm
+
+import (
+	"sync"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+)
+
+type ProgramEntry struct {
+	Location common.Location
+	Program  *interpreter.Program
+}
+
+type Programs struct {
+	lock     sync.RWMutex
+	parent   *Programs
+	programs map[common.LocationID]ProgramEntry
+}
+
+func NewPrograms(parent *Programs) *Programs {
+	return &Programs{
+		parent:   parent,
+		programs: map[common.LocationID]ProgramEntry{},
+	}
+}
+
+func (p *Programs) Get(locationID common.LocationID) *interpreter.Program {
```

Comment:
I think might be better to change this to Get(location common.Location) and we do the cast to ID here.

Reply:
Good idea, changed in 694e65499b1ac4447d4623a892353652aa5a66b8